### PR TITLE
Add Collection Object

### PR DESCRIPTION
**Pull request by andersy005**
Towards #1

```python
In [1]: import xarray as xr, xcollection

In [2]: ds = xr.tutorial.open_dataset('rasm')

In [3]: ds.attrs = {}

In [4]: c = xcollection.Collection({'foo': ds, 'bar': ds})

In [5]: c
Out[5]: 
<Collection (2 keys)>
🔑 foo
<xarray.Dataset>
Dimensions:  (time: 36, y: 205, x: 275)
Coordinates:
  * time     (time) object 1980-09-16 12:00:00 ... 1983-08-17 00:00:00
    xc       (y, x) float64 ...
    yc       (y, x) float64 ...
Dimensions without coordinates: y, x
Data variables:
    Tair     (time, y, x) float64 ...

🔑 bar
<xarray.Dataset>
Dimensions:  (time: 36, y: 205, x: 275)
Coordinates:
  * time     (time) object 1980-09-16 12:00:00 ... 1983-08-17 00:00:00
    xc       (y, x) float64 ...
    yc       (y, x) float64 ...
Dimensions without coordinates: y, x
Data variables:
    Tair     (time, y, x) float64 ...

```

### CODE REVIEW

**Review by kmpaul — COMMENTED** [review]
Why no test for __getitem__?

File: tests/test_main.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ds = xr.tutorial.open_dataset('rasm')
+
+
+@pytest.mark.parametrize('datasets', [None, {'a': ds, 'b': ds}])
+def test_init(datasets):
+    c = xcollection.Collection(datasets)
+    assert isinstance(c.datasets, dict)
+    if datasets is not None:
+        assert len(c) == len(datasets)
+        assert set(c.keys()) == set(datasets.keys())
```
Comment (dcherian) [inline tests/test_main.py:18]:
```suggestion
        assert set(c.keys()) == set(datasets.keys())
    else:
        assert len(c) == 0
```

File: tests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def test_repr():
+    c = xcollection.Collection({'bar': ds, 'foo': ds})
+    assert 'foo' in repr(c)
+    assert 'bar' in repr(c)
+
+
+@pytest.mark.parametrize('datasets', [{'a': ds, 'b': 5}, {1: ds}, {'test': ds.Tair}])
```
Comment (dcherian) [inline tests/test_main.py]: Not needed now but we *could* promote DataArrays to Datasets use `.to_dataset()`

File: xcollection/main.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    def __repr__(self):
+        unicode_key = u'\U0001F511'
+        output = ''.join(f'{unicode_key} {key}\n{repr(value)}\n\n' for key, value in self.items())
+        return f'<{type(self).__name__} ({len(self)} keys)>\n{output}'
+
+    def _repr_html_(self):
+        """
+        Return an html representation for the collection object.
+        Mainly for IPython notebook
+        """
```
Comment (mgrover1) [inline xcollection/main.py:63]: Is the plan to put something here?

Comment (andersy005) [reply xcollection/main.py:63]: Yes, it's a placeholder for the `Extend Xarray HTML wrapper ...` to-do item...

**Comment by andersy005** [comment]
> Why no test for **getitem**?

Good catch... Done!

File: xcollection/main.py (hunk 3 above)
Comment (mgrover1) [reply xcollection/main.py:63]: Sounds good!

File: tests/test_main.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    del c['a']
+    assert len(c) == 1
+    assert set(c.keys()) == {'b'}
+
+
+def test_getitem():
+    datasets = {'a': ds, 'b': ds}
+    c = xcollection.Collection(datasets)
+    assert isinstance(c['a'], xr.Dataset)
+
```
Comment (kmpaul) [inline tests/test_main.py]:
```suggestion
    assert isinstance(c['a'], xr.Dataset)
    assert c['a'] is ds
    assert c['b'] is ds

```

**Review by kmpaul — CHANGES_REQUESTED** [review]
One more suggestion...

File: tests/test_main.py (hunk 2 above)
Comment (andersy005) [reply tests/test_main.py]: :+1:

Supporting this requires a small change and it would be convenient for users. I'm going to include this shortly

**Review by kmpaul — APPROVED** [review]
Looks great!  Thanks, @andersy005!

**Comment by andersy005** [comment]
Thank you, @kmpaul, @mgrover1, @dcherian!

**Pull request merged**